packages/blade/src/components/PreviewWindow/PreviewWindow.web.tsx

```diff
@@ -0,0 +1,339 @@
+  .zoom-wrapper,
+  .zoom-content {
+    width: 100%;
+    height: 100%;
+    overflow: visible;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+  }
+  transition: cursor 0.1s ease;
+`;
+const PreviewWindow = ({
+  children,
+  onFullScreen: onFullScreenProp,
+  onZoomChange,
+  zoomScaleStep = 0.1,
+  isDragAndZoomDisabled = false,
+  initialZoom,
+  onDragChange,
+}: PreviewWindowProps): React.ReactElement => {
+  const [controlledZoom, setControlledZoom] = useControllableState({
+    onChange: onZoomChange,
+    defaultValue: initialZoom ?? 1,
+  });
+  const [isFullScreen, setIsFullScreen] = useState(false);
+  const [isDragging, setIsDragging] = useState(false);
+  const containerRef = React.useRef<HTMLDivElement>(null);
+
+  const handleTransformed = ({
+    state,
+  }: {
+    state: { scale: number; positionX: number; positionY: number };
+  }): void => {
+    const { scale, positionX, positionY } = state;
+    setControlledZoom(() => scale);
+    onDragChange?.({ x: positionX, y: positionY });
+  };
+
+  const handleFullScreen = useCallback(() => {
+    if (!document.fullscreenElement) {
+      containerRef.current?.requestFullscreen().catch((err) => {
+        console.error(`Error attempting to enable fullscreen: ${err.message}`);
+      });
+      setIsFullScreen(true);
+      onFullScreenProp?.();
+    } else {
+      document.exitFullscreen().catch((err) => {
+        console.error(`Error attempting to exit fullscreen: ${err.message}`);
+      });
+      setIsFullScreen(false);
+      onFullScreenProp?.();
+    }
+  }, [onFullScreenProp]);
+
+  // Handle keyboard shortcut for fullscreen
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent): void => {
+      if ((event.metaKey || event.ctrlKey) && event.key === 'f') {
+        event.preventDefault();
+        handleFullScreen();
+      }
+    };
+
+    document.addEventListener('keydown', handleKeyDown);
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [handleFullScreen]);
+
+  // this is added to handle the fullscreen change if user exits fullscreen using the escape key, or browser's exit fullscreen button
+  useEffect(() => {
+    const handleFullScreenChange = (): void => {
+      setIsFullScreen(!!document.fullscreenElement);
+    };
+
+    document.addEventListener('fullscreenchange', handleFullScreenChange);
+    return () => {
+      document.removeEventListener('fullscreenchange', handleFullScreenChange);
+    };
+  }, []);
+
+  // filter out preview header, preview body, preview footer separately using componentId
+  const previewHeader = React.Children.toArray(children)
+    .filter((child) => getComponentId(child as React.ReactElement) === MetaConstants.PreviewHeader)
+    .map((child) => {
+      if (React.isValidElement(child)) {
+        return React.cloneElement(child, {
+          // @ts-expect-error
+          _onFullScreen: handleFullScreen,
```

Comment:
Either way is fine but if you want you can also use the same PreviewWindowContext for keeping onFullScreen as well right?

Reply:
will go with this way only..